### PR TITLE
docs: add rotation guide

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Adds rotation guide

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

File: docs/rotation.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+### 2. Deploy another instance of KMS plugin with new key
+
+To rotate the encrypt/decrypt key in the cluster, you'll need to run 2 kms plugin pods simultaneously listening on different unix sockets before making the transition.
+
+For all Kubernetes master nodes, add the static pod manifest to `/etc/kubernetes/manifests`
+
+To rotate the encrypt/decrypt key in the cluster, you'll need to run 2 kms plugin pods simultaneously listening on different unix sockets before making the transition.
+
+For all Kubernetes master nodes, add the static pod manifest to `/etc/kubernetes/manifests`
```
Comment (sozercan) [inline docs/rotation.md:20]: duplicate

File: docs/rotation.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
+# Rotating KMS key
+
+This guide demonstrates steps required to update your cluster to use a new KMS key for encryption.
+
+> NOTE: Ensure to read the Kubernetes documentation on [Rotating a decryption key](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#rotating-a-decryption-key) before proceeding with the guide.
+
+### 1. Generate a new key or rotate the existing key
+
+* If this is a new key in a different keyvault, then give the cluster identity permissions to access the keys in keyvault. Refer to [doc](https://github.com/Azure/kubernetes-kms/blob/master/docs/manual-install.md#2-give-the-cluster-identity-permissions-to-access-the-keys-in-keyvault) for details.
```
Comment (sozercan) [inline docs/rotation.md:9]: use relative path instead of static url?

File: docs/rotation.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        - containerPort: 8788                                   # Must match the value defined in --healthz-port
+            protocol: TCP
+        livenessProbe:
+        httpGet:
+            path: /healthz                                      # Must match the value defined in --healthz-path
+            port: 8788                                          # Must match the value defined in --healthz-port
+        failureThreshold: 2
+        periodSeconds: 10
+        resources:
+        requests:
```
Comment (sozercan) [inline docs/rotation.md:58]: indentation issue here

File: docs/rotation.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+### 3. Add the new provider to encryption configuration in `/etc/kubernetes/manifests/encryptionconfig.yaml`
+
+```yaml
+kind: EncryptionConfiguration
+apiVersion: apiserver.config.k8s.io/v1
+resources:
+    - resources:                                            # List of kubernetes resources that will be encrypted in etcd using the KMS plugin
+        - secrets
+    providers:
```
Comment (sozercan) [inline docs/rotation.md:105]: indentation issue here, should be under resources

File: docs/rotation.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+### 5. Switch the order of provider in the encryption config
+
+```yaml
+kind: EncryptionConfiguration
+apiVersion: apiserver.config.k8s.io/v1
+resources:
+    - resources:                                            # List of kubernetes resources that will be encrypted in etcd using the KMS plugin
+        - secrets
+    providers:
```
Comment (sozercan) [inline docs/rotation.md:130]: same re: indentation

File: docs/rotation.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+            name: azurekmsprovider2
+            endpoint: unix:///opt/azurekms2.socket          # This endpoint must match the value defined in --listen-addr for the KMS plugin using new key
+            cachesize: 1000
+```
+
+### 4. Restart all `kube-apiserver`
+
+* Proceed to the next step if using a single `kube-apiserver`
+* If using multi-master, restart the `kube-apiserver` to ensure each server can still decrypt using the new key in the encryption config.
+  * To validate the decryption still works, run `kubectl get secret <secret name> -o yaml` with one of the existing secrets to confirm the data is returned and is valid.
```
Comment (sozercan) [inline docs/rotation.md:120]:
```suggestion
* To validate the decryption still works, run `kubectl get secret <secret name> -o yaml` with one of the existing secrets to confirm the data is returned and is valid.
```

File: docs/rotation.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+            endpoint: unix:///opt/azurekms2.socket          # This endpoint must match the value defined in --listen-addr for the KMS plugin using new key
+            cachesize: 1000
+        # kms provider with old key
+        - kms:
+            name: azurekmsprovider
+            endpoint: unix:///opt/azurekms.socket           # This endpoint must match the value defined in --listen-addr for the KMS plugin using old key
+            cachesize: 1000
+```
+
+### 6. Restart all `kube-apiserver`
```
Comment (sozercan) [inline docs/rotation.md:143]: same as number 4?

File: docs/rotation.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+Now that all the secrets have been re-encrypted with the new key, we can safely remove the old kms provider from the encryption configuration.
+
+```yaml
+kind: EncryptionConfiguration
+apiVersion: apiserver.config.k8s.io/v1
+resources:
+    - resources:                                            # List of kubernetes resources that will be encrypted in etcd using the KMS plugin
+        - secrets
+    providers:
```
Comment (sozercan) [inline docs/rotation.md:169]: same re: indentation

File: docs/rotation.md (hunk 7 above)
Comment (sozercan) [reply docs/rotation.md:143]: do they need to refer to same instructions in number 4?

Comment (aramase) [reply docs/rotation.md]: Yeah, added a reference to step 4.

**Review by sozercan — APPROVED** [review]
Thanks! LGTM

**Pull request merged**